apps/src/lib/tools/jsdebugger/DebugConsole.jsx

```diff
@@ -149,13 +158,35 @@ export default connect(
           );
         } else if (this.props.isAttached) {
           try {
-            const result = this.props.evalInCurrentScope(input);
-            this.appendLog('< ' + String(result));
+            if (experiments.isEnabled('react-inspector')) {
+              // wrapping input with parens for objects before the evalInCurrestScope
+              // ensures objects are returned correctly consistently
+              let result = this.props.evalInCurrentScope(
+                input[0] === '{' && input[input.length - 1] === '}'
+                  ? `(${input})`
+                  : input
+              );
+              result = this.props.jsInterpreter.interpreter.marshalInterpreterToNative(
+                result
+              );
+              this.appendLog({
+                output: result,
+                fromConsoleLog: false,
+                undefinedInput: input === 'undefined' ? true : false
+              });
+            } else {
+              const result = this.props.evalInCurrentScope(input);
+              this.appendLog('< ' + String(result));
+            }
           } catch (err) {
-            this.appendLog('< ' + String(err));
+            experiments.isEnabled('react-inspector')
+              ? this.appendLog({output: String(err), fromConsoleLog: false})
+              : this.appendLog('< ' + String(err));
           }
         } else {
-          this.appendLog('< (not running)');
+          experiments.isEnabled('react-inspector')
+            ? this.appendLog({output: '(not running)', fromConsoleLog: false})
```

Comment:
You don't need to explicitly set fromConsoleLog to false (here and 183). 
Reason: If you don't set output.fromConsoleLog, it will evaluate to undefined which will evaluate to false.

Reply:
Removed fromConsoleLog from line 174, 183, and 188!